File: Database/IDatabaseObjectLockable.vb

' ___________________________________________________
'
'  © Hi-Integrity Systems 2010. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On 

''' --------------------------------------------------------------------------------
''' <summary>
''' Specifies the interface of all classes that are considered lockable by the library.
''' See DatabaseObjectLockable, DatabaseObjectUsingAttributesLockable for further
''' details.
''' </summary>
''' --------------------------------------------------------------------------------
Public Interface IDatabaseObjectLockable

    Sub Lock()
    Sub UnLock()
    ReadOnly Property LockedByUserID() As String
    ReadOnly Property IsLocked() As Boolean
    ReadOnly Property IsLockedByCurrentUser() As Boolean

End Interface

File: Database/IDatabaseObjectLockable.cs
// ___________________________________________________
//
//  © Hi-Integrity Systems 2010. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// ___________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects
{
	/// --------------------------------------------------------------------------------
	/// <summary>
	/// Specifies the interface of all classes that are considered lockable by the library.
	/// See DatabaseObjectLockable, DatabaseObjectUsingAttributesLockable for further
	/// details.
	/// </summary>
	/// --------------------------------------------------------------------------------
	public interface IDatabaseObjectLockable
	{
		void Lock();
		void UnLock();
		string LockedByUserID { get; }
		bool IsLocked { get; }
		bool IsLockedByCurrentUser { get; }
	}
}
